### ISSUE TITLE
Error in Yii2 before test

### ISSUE THREAD

**Issue by fanfromzp** [Codeception/phpunit-wrapper#24]
Warning: Declaration of Codeception\PHPUnit\ResultPrinter\UI::printDefect(PHPUnit\Framework\TestFailure $defect, $count): void should be compatible with PHPUnit\TextUI\ResultPrinter::printDefect(PHPUnit\Framework\TestFailure $defect, int $count): void

Need to change in UI.php
protected function printDefect(\PHPUnit\Framework\TestFailure $defect, $count)
to
protected function printDefect(\PHPUnit\Framework\TestFailure $defect, int $count)

**Comment by roslov** [comment]
Because of this I cannot use Codeception 2.4.1 :-(

**Comment by Naktibalda** [comment]
@roslov  If you want to get Codeception 2.4.1 working with PHPUnit 7 now, add constraint `"phpunit/phpunit": "7.0.*"` to require-dev section of your composer.json file.

**Comment by roslov** [comment]
@Naktibalda It didn’t work.
Now I have an error `[PHPUnit\Framework\Exception] Undefined variable: response`.

```
/var/www/html/vendor/codeception/codeception/src/Codeception/Subscriber/ErrorHandler.php:83
/var/www/html/vendor/codeception/codeception/src/Codeception/Lib/Connector/Yii2.php:154
/var/www/html/vendor/symfony/browser-kit/Client.php:312
/var/www/html/vendor/codeception/codeception/src/Codeception/Lib/InnerBrowser.php:193
/var/www/html/vendor/codeception/codeception/src/Codeception/Module/Yii2.php:670
/var/www/html/vendor/codeception/codeception/src/Codeception/Lib/InnerBrowser.php:125
/var/www/html/vendor/codeception/codeception/src/Codeception/Module/REST.php:597
/var/www/html/vendor/codeception/codeception/src/Codeception/Module/REST.php:459
/var/www/html/vendor/codeception/codeception/src/Codeception/Step.php:264
/var/www/html/vendor/codeception/codeception/src/Codeception/Scenario.php:72
...
```

Codeception 2.4.0 works good but with well known `Uncaught Error: Call to undefined method Codeception\Test\Descriptor::getTestDataSetIndex()` :-(

**Comment by SamMousa** [comment]
@roslov The undefined variable error you're getting is something I've already fixed in codeception; could you confirm that it works and if not make an issue there? Feel free to @ ping me in it.

The composer constraint you can use is `2.4.x-dev`

**Comment by roslov** [comment]
@SamMousa, with `phpunit/phpunit:7.0.*` and `codeception/codeception:2.4.x-dev` a new bug appeared:

```
[yii\db\Exception] SQLSTATE[HY000] [1040] Too many connections
```

I haven’t seen this bug for a long time.

I have ~1.5K tests in my project (including API, acceptance, functional and unit tests).

If I revert back to `2.4.0` it works good. Looks like changes `Multiple databases are now supported` or `More reliable application state before and during test execution` of `2.4.1` broke this again.

**Comment by SamMousa** [comment]
@roslov You are right, this is fixed in an open PR.
The cause is that PDO objects are cached but the cache was actually never filled..

**Comment by roslov** [comment]
@SamMousa, will it be fixed in `2.4.2`?

**Comment by SamMousa** [comment]
I imagine yes.
